File: cocmd_old/cocmd_cli_py/core/models/script_model.py
from dataclasses import dataclass, field
from cocmd.core.os import OS
from cocmd.utils.io import DictLoader
from typing import List, Optional
from enum import Enum


class StepRunnerType(Enum):
    SHELL = "shell"
    MARKDOWN = "markdown"
    PYTHON = "python"
    COCMD_SCRIPT = "cocmd_script"
    LINK = "link"


@dataclass
class StepModel(DictLoader):
    runner: StepRunnerType
    content: str
    title: str


@dataclass
class ScriptModel(DictLoader):
    steps: List[StepModel]
    env: Optional[OS] = field(default=OS.ANY)
    description: Optional[str] = field(default=None)

File: cocmd/src/core/models/script_model.rs
use serde_derive::{Deserialize, Serialize};

use crate::utils::sys::OS;

#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Hash, Clone)]
pub enum StepRunnerType {
    #[serde(alias = "shell", alias = "SHELL")]
    SHELL,
    #[serde(alias = "markdown", alias = "MARKDOWN")]
    MARKDOWN,
    #[serde(alias = "python", alias = "PYTHON")]
    PYTHON,
    #[serde(alias = "link", alias = "LINK")]
    LINK,
    #[serde(alias = "cocmd", alias = "COCMD")]
    COCMD,
}

#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Hash, Clone)]
pub struct StepModel {
    pub runner: StepRunnerType,
    pub content: Option<String>,
    pub file: Option<String>,
    pub title: String,
}

#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Hash, Clone)]
pub struct ScriptModel {
    pub steps: Vec<StepModel>,
    pub env: Option<OS>,
    pub description: Option<String>,
}
